src/IW/Effects/Download.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
+
+{- | This module contains the class definitiion of @MonadDownload@ and 
+an instance of @MonadDownload@ for the @App@ monad. Instances of 
+@MonadDownload@ have a @downloadFile@ action for downloading files from a given URL.
+-}
+
+module IW.Effects.Download
+       ( MonadDownload (..)
+        
+       -- * Internals
+       , downloadFileImpl
+       ) where
+
+import Network.HTTP.Client (Manager, Response (..), httpLbs)
+import Network.HTTP.Types (Status (..))
+
+import IW.App (App, Has, WithError, grab, throwError, notFound)   
+import IW.Core.Url (Url (..)) 
+
+-- | Describes a monad that can download files from a given URL.
+class Monad m => MonadDownload m where
+    downloadFile :: Url -> m ByteString
+
+instance MonadDownload App where 
+    downloadFile = downloadFileImpl
```

Comment:
I think it should be possible to write couple tests for this function. Our tests should be able to download `.cabal` from `issue-wanted` itself, I guess

Reply:
@chshersh Sounds good :+1: